rnd/autogpt_libs/autogpt_libs/logging/config.py

```diff
@@ -0,0 +1,199 @@
+    )
+
+    enable_console_logging: bool = Field(
+        default=True,
+        description="Enable logging to console",
+        validation_alias="ENABLE_CONSOLE_LOGGING",
+    )
+
+    enable_json_logging: bool = Field(
+        default=False,
+        description="Enable logging to JSON file",
+        validation_alias="ENABLE_JSON_LOGGING",
+    )
+
+    # File output
+    log_dir: Path = Field(
+        default=LOG_DIR,
+        description="Log directory",
+    )
+
+    model_config = SettingsConfigDict(
+        env_prefix="",
+        env_file=".env",
+        env_file_encoding="utf-8",
+        extra="ignore",
+    )
+
+    @field_validator("level", mode="before")
+    @classmethod
+    def parse_log_level(cls, v):
+        if isinstance(v, str):
+            v = v.upper()
+            if v not in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]:
+                raise ValueError(f"Invalid log level: {v}")
+            return v
+        return v
+
+
+def configure_logging(
+        force_cloud_logging: bool = False
+) -> None:
+    """Configure the native logging module based on the LoggingConfig settings.
+
+    This function sets up logging handlers and formatters according to the
+    configuration specified in the LoggingConfig object. It supports various
+    logging outputs including console, file, cloud, and JSON logging.
+
+    The function uses the LoggingConfig object to determine which logging
+    features to enable and how to configure them. This includes setting
+    log levels, log formats, and output destinations.
+
+    No arguments are required as the function creates its own LoggingConfig
+    instance internally.
+
+    Note: This function is typically called at the start of the application
+    to set up the logging infrastructure.
+    """
+
+    config = LoggingConfig()
+
+    log_handlers: list[logging.Handler] = []
+
+    # Cloud logging setup
+    if config.enable_cloud_logging or force_cloud_logging:
+        try:
+            import google.cloud.logging
+            from google.cloud.logging.handlers import CloudLoggingHandler
+            from google.cloud.logging_v2.handlers.transports.sync import SyncTransport
+
+            client = google.cloud.logging.Client()
+            cloud_handler = CloudLoggingHandler(
+                client,
+                name="autogpt_logs",
+                transport=SyncTransport,
+            )
+            cloud_handler.setLevel(config.level)
+            cloud_handler.setFormatter(StructuredLoggingFormatter())
+            log_handlers.append(cloud_handler)
+            print("Cloud logging enabled")
+        except ImportError:
+            print(
+                "Cloud logging requested but google-cloud-logging is not installed. "
+                "Please install it with 'pip install google-cloud-logging'."
+            )
+        except Exception as e:
+            print(f"Failed to set up cloud logging: {str(e)}")
```

Comment:
No need to catch this; failure to set up logging is catastrophic and should lead to termination of the application

Reply:
agreed